lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/CommunicationGroupClient.cs

```diff
@@ -175,5 +177,18 @@ public IScatterReceiver<T> GetScatterReceiver<T>(string operatorName)
 
             return (T) op;
         }
+
+        /// <summary>
+        /// Call each Operator to easure all the nodes in the topology group has been registered
+        /// </summary>
+        void ICommunicationGroupClientInternal.WaitingForRegistration()
+        {
+            foreach (var op in _operators.Values)
+            {
+                var methods = op.GetType().GetMethods(BindingFlags.NonPublic | BindingFlags.Instance);
+                var queriedMethods = methods.Where(e => e.Name.Equals("Org.Apache.REEF.Network.Group.Operators.IGroupCommOperatorInternal.WaitForRegistration")).ToArray();
+                queriedMethods[0].Invoke(op, null);
```

Comment:
Is this neccesary? We might make `IGroupCommOperatorInternal` extend the `IGroupCommOperator` interface, right? And in that case, we can call the method directly? Or am I missing something?